### PR TITLE
[WIP]: Add maistra istio presubmit jobs

### PR DESCRIPTION
**Pull request by yxun**
Signed-off-by: Yuanlin <yuanlin.xu@redhat.com>

This PR adds OpenShift Service Mesh Maistra component jobs and tide configs

### CODE REVIEW

**Comment by yxun** [comment]
Could I get a review or /ok-to-test comment for adding OpenShift Service Mesh jobs ?
If there is any test failure, I can update the PR with additional changes. Thanks.

File: ci-operator/jobs/openshift-ossm/maistra/openshift-ossm-maistra-postsubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+    max_concurrency: 1
+    spec:
+      containers:
+      - image: "quay.io/maistra-dev/maistra-builder:2.0"
+        imagePullPolicy: Always
+        command:
+        - entrypoint
+        - make
+        - maistra-proxy-builder.push
+        securityContext:
```
Comment (yxun) [inline ci-operator/jobs/openshift-ossm/maistra/openshift-ossm-maistra-postsubmits.yaml:42]: Hello, CI maintainer, 
Is there a way of running a privileged pod in an OpenShift CI environment ?
We use that pod for running a dockerd and upstream integration tests. 
Thanks,

Comment (smg247) [reply ci-operator/jobs/openshift-ossm/maistra/openshift-ossm-maistra-postsubmits.yaml:42]: We do have precedent for doing it like this. Also for adding `runAsUser: 0`. Would that satisfy your needs?

File: ci-operator/jobs/openshift-ossm/maistra/openshift-ossm-maistra-postsubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,545 @@
+postsubmits:
```
Comment (droslean) [inline ci-operator/jobs/openshift-ossm/maistra/openshift-ossm-maistra-postsubmits.yaml:1]: `ci-operator` has a way to build and push your images. You should add them in your ci-operator configuration and let the generated postsubmit only. Try to check https://docs.ci.openshift.org/docs/architecture/ci-operator/ to get more context about how ci-operator works.

Comment (yxun) [reply ci-operator/jobs/openshift-ossm/maistra/openshift-ossm-maistra-postsubmits.yaml:1]: ok, I will update my PR and thanks for your suggestion.

Comment (yxun) [reply ci-operator/jobs/openshift-ossm/maistra/openshift-ossm-maistra-postsubmits.yaml:1]: Hi @droslean , I updated my PR using the ci-operator. Could you have a review ?
Is there a doc or explanation about how the ci-operator watches a github repo and branch ?
Thanks,

File: ci-operator/jobs/openshift-ossm/maistra/openshift-ossm-maistra-postsubmits.yaml [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+  maistra/envoy:
```
Comment (droslean) [inline ci-operator/jobs/openshift-ossm/maistra/openshift-ossm-maistra-postsubmits.yaml:2]: Please remove all of the presubmits. Try to `make jobs` in order to generate the correct one. In order to have postsubmits you will need to add a `promotion` configuration in the ci-operator config.

File: ci-operator/jobs/openshift-ossm/maistra/openshift-ossm-maistra-postsubmits.yaml (hunk 2 above)
Comment (yxun) [reply ci-operator/jobs/openshift-ossm/maistra/openshift-ossm-maistra-postsubmits.yaml:1]: Sure,
For example, if I need to set up a presubmit test for
https://github.com/maistra/istio
branch maistra-2.2

I am adding the following in the ci-operator config:
https://github.com/openshift/release/blob/3f92cabcd546fd1f858699cf59da80aa84456020/ci-operator/config/maistra/istio/maistra-istio-2.2.yaml#L231
Is that correct ? I assume the ci-operator job will clone the github maistra/istio repo and checkout the branch.

**Comment by yxun** [comment]
This PR is ready for review. I also rebased the recent changes from master branch. Thanks.

**Comment by yxun** [comment]
I will add the `cluster_profile` and `workflow` fields in my ci-operator config. It looks like the prow config test failed with the following error.
""" {"component":"checkconfig","error":"dialing: google: could not find default credentials...."""

**Comment by yxun** [comment]
Should I create a separate PR for the "mapping_supplemental_ci_images_ci" change first ?

**Comment by yxun** [comment]
This PR depends on https://github.com/openshift/release/pull/30939

**Pull request merged**